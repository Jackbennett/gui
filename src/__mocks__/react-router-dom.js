import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
