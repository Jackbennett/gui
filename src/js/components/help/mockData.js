export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
