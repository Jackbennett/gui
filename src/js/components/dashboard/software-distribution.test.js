import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import SoftwareDistribution from './software-distribution';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('Devices Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      app: {
        features: {
          isEnterprise: false
        }
      },
      deployments: {
        deploymentDeviceLimit: 5000
      },
      devices: {
        byId: {},
        byStatus: {
          accepted: {
            total: 0,
            deviceIds: []
          },
          active: {
            total: 0
          },
          inactive: {
            total: 0
          },
          pending: {
            total: 0
          }
        },
        filteringAttributes: {
          inventoryAttributes: []
        },
        groups: {
          byId: {}
        }
      },
      users: {
        byId: {},
        currentUser: null,
        globalSettings: {},
        onboarding: { complete: false },
        showHelptips: true
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <SoftwareDistribution getAllDevicesByStatus={jest.fn()} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
