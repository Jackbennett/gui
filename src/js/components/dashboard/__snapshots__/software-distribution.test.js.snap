// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices Component renders correctly 1`] = `
<div>
  <div
    class="flexbox centered"
  >
    <p
      class="info icon "
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        style="vertical-align: middle; margin: 0px 6px 4px 0px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      With a more advanced commercial Mender plan you get access to actionable insights into the devices you are updating with Mender. Get an overview of the Mender plans on the 
      <a
        href="https://mender.io/plans/features"
        rel="noopener noreferrer"
        target="_blank"
      >
        features page
      </a>
      .
    </p>
  </div>
</div>
`;
